src/sql/src/plan/statement.rs

```diff
@@ -971,12 +971,28 @@ fn handle_create_type(
     let mut ids = vec![];
     for key in option_keys {
         let item_name = match with_options.remove(&key.to_string()) {
-            Some(SqlOption::Value {
-                value: Value::String(val),
-                ..
-            }) => ObjectName::unqualified(&val),
-            Some(SqlOption::ObjectName { object_name, .. }) => object_name,
-            Some(_) => bail!("{} must be a string or identifier", key),
+            Some(SqlOption::DataType { data_type, .. }) => match data_type {
+                DataType::Other { name, typ_mod } => {
+                    if !typ_mod.is_empty() {
+                        bail!(
```

Comment:
Why can't we accept type modifiers here?

Reply:
At the highest level, we only store a type's OID, which isn't correlated to any type modifiers.

The only type modifier we actually use is the one on `numeric`/`decimal`, to specify the precision and scale; however, all `numeric` types share the same OID, which means no matter what you specify you end up with the default value––I have this tracked in #4884.